src/dotnet/UI.Blazor/Components/VisualMediaViewerModal/visual-media-viewer.ts

```diff
@@ -172,20 +247,54 @@ export class VisualMediaViewer {
-        this.footerTop = this.round(parent.querySelector('.image-viewer-footer').getBoundingClientRect().top);
-        const imageRect = this.imageViewer.getBoundingClientRect();
-        const imageTop = this.round(imageRect.top);
-        const imageBottom = this.round(imageRect.bottom);
-        const imageLeft = this.round(imageRect.left);
-        const imageRight = this.round(imageRect.right);
-        if (imageTop < this.headerBottom || imageBottom > this.footerTop || imageLeft < 0 || imageRight > window.innerWidth) {
-            this.getImageAndMouseX(event, imageRect);
-            this.getImageAndMouseY(event, imageRect);
-            window.addEventListener('pointermove', this.onPointerMove);
-        } else {
-            preventDefaultForEvent(event);
+        if (event.pointerType == 'mouse' && this.imageViewer.contains(event.target as HTMLElement)) {
+            const parent = this.imageViewer.parentElement;
+            this.headerBottom = this.round(parent.querySelector('.image-viewer-header').getBoundingClientRect().bottom);
+            this.footerTop = this.round(parent.querySelector('.image-viewer-footer').getBoundingClientRect().top);
+            const imageRect = this.imageViewer.getBoundingClientRect();
+            const imageTop = this.round(imageRect.top);
+            const imageBottom = this.round(imageRect.bottom);
+            const imageLeft = this.round(imageRect.left);
+            const imageRight = this.round(imageRect.right);
+            if (imageTop < this.headerBottom || imageBottom > this.footerTop || imageLeft < 0 || imageRight > window.innerWidth) {
+                this.getImageAndMouseX(event, imageRect);
+                this.getImageAndMouseY(event, imageRect);
+                window.addEventListener('pointermove', this.onPointerMove);
+            } else {
+                preventDefaultForEvent(event);
+            }
+        } else if (event.pointerType == 'touch') {
+            window.addEventListener('pointermove', this.onTouchableMove);
+            this.image.style.touchAction = 'none';
+            this.imageViewer.style.touchAction = 'none';
+            this.overlay.style.touchAction = 'none';
+            this.points.push(event);
+            this.startImageRect = this.imageViewer.getBoundingClientRect();
+            this.prevImageRect = this.startImageRect;
+            if (this.points.length === 2) {
+                this.startDistance = this.getDistance();
+                this.prevDistance = this.getDistance();
+            } else if (this.points.length === 1) {
+                this.startY = event.pageY;
+                this.startX = event.pageX;
+                this.prevY = this.startY;
+                this.prevX = this.startX;
+                let target = event.target as HTMLElement;
+                if (this.imageViewer.contains(target)) {
+                    setTimeout(() => {
```

Comment:
Why do you need setTimeout here? Can you explain why you call footerHeaderToggle? And why does it require condition `this.points.length < 2 && !this.movementStart` ? I see logical expression but I can't guess why it means.

Reply:
Yep, zero comments in code.